### PR TITLE
Add model data pruner job

### PR DESCRIPTION
**Pull request by conbrad**
Adds a model data pruner job that:
- Checks for the earliest model run record in `weather_station_model_predictions`, and attempts to delete 14 days worth of records starting at that date, if that date range is before the range of `[today - 19 days, today]`
- If the date range starts before `today - 19 days` but the end of the range doesn't, the range will be 'clamped' to `today - 19 days`
- Runs every 10 minutes * 24 14 day periods a year = 1 year of record pruning in 4 hours
# Test Links:
[Landing Page](https://wps-pr-2952.apps.silver.devops.gov.bc.ca/)
[MoreCast 2.0](https://wps-pr-2952.apps.silver.devops.gov.bc.ca/morecast-2)
[Percentile Calculator](https://wps-pr-2952.apps.silver.devops.gov.bc.ca/percentile-calculator)
[MoreCast](https://wps-pr-2952.apps.silver.devops.gov.bc.ca/morecast)
[C-Haines](https://wps-pr-2952.apps.silver.devops.gov.bc.ca/c-haines)
[FireBat](https://wps-pr-2952.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator)
[FireBat bookmark](https://wps-pr-2952.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator?s=266&f=c5&c=NaN&w=20,s=286&f=c7&c=NaN&w=16,s=1055&f=c7&c=NaN&w=NaN,s=305&f=c7&c=NaN&w=NaN,s=344&f=c5&c=NaN&w=NaN,s=346&f=c7&c=NaN&w=NaN,s=328&f=c7&c=NaN&w=NaN,s=1399&f=c7&c=NaN&w=NaN,s=334&f=c7&c=NaN&w=NaN,s=1082&f=c3&c=NaN&w=NaN,s=388&f=c7&c=NaN&w=NaN,s=309&f=c7&c=NaN&w=16,s=306&f=c7&c=NaN&w=NaN,s=1029&f=c7&c=NaN&w=NaN,s=298&f=c7&c=NaN&w=NaN,s=836&f=c7&c=NaN&w=NaN,s=9999&f=c7&c=NaN&w=NaN)
[Auto Spatial Advisory (ASA)](https://wps-pr-2952.apps.silver.devops.gov.bc.ca/auto-spatial-advisory)
[HFI Calculator](https://wps-pr-2952.apps.silver.devops.gov.bc.ca/hfi-calculator)

### CODE REVIEW

File: api/app/jobs/model_data_pruner.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+            range_start, range_end = get_model_prune_range(now, start, end)
+            logger.info(f"Pruning model data from {range_start} to {range_end}")
+            prune_model_station_predictions_between(session, start, end)
+    except CompletedWithSomeExceptions:
+        logger.warning('completed processing with some exceptions')
+        sys.exit(os.EX_SOFTWARE)
+    except Exception as exception:
+        # We catch and log any exceptions we may have missed.
+        logger.error('unexpected exception processing', exc_info=exception)
+        rc_message = ':poop: Encountered error retrieving GFS model data from NOAA'
```
Comment (dgboss) [inline api/app/jobs/model_data_pruner.py]: Copy - paste remnant on line 36

File: api/app/utils/time.py [hunk 2]
```diff
@@ -120,3 +120,15 @@ def get_utc_datetime(input_datetime: datetime):
 
 def get_days_from_range(start_time: datetime, end_time: datetime) -> List[datetime]:
     return [start_time + timedelta(days=x) for x in range((end_time - start_time).days + 1)]
+
+
+def get_model_prune_range(now: datetime, start: datetime, end: datetime):
+    # only allow pruning up until 19 days before today
+    max_end = now - timedelta(days=19)
```
Comment (dgboss) [inline api/app/utils/time.py:127]: `machine_learning.py` also uses the number 19, could we define a constant somewhere or set things up to be read from a config map? Maybe suitable as a follow on ticket?

Comment (dgboss) [reply api/app/utils/time.py:127]: Should we consider defaulting to 21 days to be safe and consistent with `apply_data_retention_policy()` in `common_model_fetchers.py`?

**Review by dgboss — APPROVED** [review]

File: api/app/utils/time.py (hunk 2 above)
Comment (conbrad) [reply api/app/utils/time.py]: I pulled out the 21 day timedelta as constant and used it across this job and `apply_data_retention_policy()` in: https://github.com/bcgov/wps/pull/2952/commits/69c5fa021894c13b8ff7eaf111fcc151ab7c6bfd

**Pull request merged**